lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3329,7 +3329,17 @@ private function isIdentifierEquals($entity1, $entity2)
             ? $this->entityIdentifiers[$oid2]
             : $this->identifierFlattener->flattenIdentifier($class, $class->getIdentifierValues($entity2));
 
-        return $id1 === $id2 || implode(' ', $id1) === implode(' ', $id2);
+        return $id1 === $id2 || $this->hashIdentifier($id1) === $this->hashIdentifier($id2);
+    }
+
+    /**
+     * @param array $identifier
+     *
+     * @return string
+     */
+    private function hashIdentifier(array $identifier)
+    {
+        return serialize(array_values(array_map('strval', $identifier)));
```

Comment:
Btw I still have a lot of projects using Doctrine 1 and it's a lot worse than Doctrine 2 performance and memory wise and still acceptable (at least performance wise) as long as the database server is strong enough to deliver the stuff. Even with 100+ entities. So I really don't see a problem with this.